### ISSUE TITLE
[Feature] Can you provide a trace viewer NPM package

### ISSUE THREAD

**Issue by xflihaibo** [microsoft/playwright#9883]
Hello

Can you provide a Trace Viewer NPM package,

We are working on an open source test GUI application Etest x wants to add this function

https://onepiece-smile.github.io/docs/#/./application/ete

thank you

**Comment by kachkaev** [comment]
@xflihaibo have you find any solution? I am looking for ways to do something like:

```js
const myTrace = await openTrace("./path-to-trace.zip");
const actions = await myTrace.listActions();
const lastSnapshotDom = await actions.at(-1)?.getDom();
```

I haven’t found any public API to do read traces in Node.js so far.

**Comment by kachkaev** [comment]
Seeing some positive changes in the repo structure: `trace-viewer` is a [separate package](https://github.com/microsoft/playwright/tree/main/packages/trace-viewer) now 👀

It’d be great to see a variant of it published on NPM! 🙏

**Comment by rwoll** [comment]
I'm coming here after reading https://github.com/microsoft/playwright/issues/16367#issuecomment-1209892073. Given we haven't seen much activity/requests here, it's unlikely we'll be resolving this soon. However, would you be open to contributing a PR? If so—before starting any work—I can raise with the team and see if it's something Playwright would include. Thanks!

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.
